internal/hcl/module.go

```diff
@@ -24,6 +29,57 @@ type Module struct {
 	RootPath   string
 	ModulePath string
 
-	Modules []*Module
-	Parent  *Module
+	Modules  []*Module
+	Parent   *Module
+	Warnings []Warning
+}
+
+// WarningCode is used to delineate warnings across Infracost.
+type WarningCode int
+
+const (
+	WarningMissingVars WarningCode = iota + 1
+)
+
+// Warning holds information about non-critical errors that occurred within a module evaluation.
+type Warning struct {
+	Code  WarningCode
+	Title string
+	Data  interface{}
+
+	// FriendlyMessage should be used to display a readable message to the CLI user.
+	FriendlyMessage string
+}
+
+// NewMissingVarsWarning returns a Warning using the WarningMissingVars error code. It expects that vars
+// is a list of Terraform variables that cannot be found in the evaluation context.
+func NewMissingVarsWarning(vars []string) Warning {
+	return Warning{
+		Code:  WarningMissingVars,
+		Title: "Missing Terraform vars",
+		Data:  vars,
+		FriendlyMessage: fmt.Sprintf(
+			"Input values were not provided for following Terraform variables: %s. %s",
+			joinQuotes(vars),
+			"Use --terraform-var-file or --terraform-var to specify them.",
+		),
+	}
+}
+
+func joinQuotes(elems []string) string {
+	switch len(elems) {
+	case 0:
+		return ""
+	case 1:
+		return fmt.Sprintf("%q", elems[0])
+	}
```

Comment:
Why is the `case 1` needed, is it not already handled below?

An alternative would be to loop through and quote the strings and then call `strings.Join(arr, ", ")`.

Reply:
no, this will panic with the range index, but I'll change to your suggestion using `strings.Join` as it's easier to understand

Reply:
updated in https://github.com/infracost/infracost/pull/2037/commits/82b9d735eeb1f8e806cd8056a715e483061ca76c